spock-core/src/main/java/org/spockframework/mock/EmptyOrDummyResponse.java

```diff
@@ -90,6 +113,7 @@ public Object respond(IMockInvocation invocation) {
     }
 
     if (returnType == OPTIONAL) return ReflectionUtil.invokeMethod(null, optionalEmptyMethod);
+    if (returnType == COMPLETABLE_FUTURE) return ReflectionUtil.invokeMethod(null, completableFutureCompletedFutureMethod, (Object) null);
```

Comment:
Well if you look into it, you could replace `IMockInvocation` for `createDummy` with the three components that are used in there. So a single level handling of generics is definitely possible however you were right that it opens up problems with nested types such as `CompletableFuture<CompletableFuture<CompletableFuture<String>>>` handling those adds much more compexity.